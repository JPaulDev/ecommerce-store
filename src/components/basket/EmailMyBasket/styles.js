import styled, { css } from 'styled-components';

export const Container = styled.div`
  background-color: var(--color-grey-15);
  border: 1px solid var(--color-grey-12);
  border-radius: var(--border-radius-2xl);
  font-size: var(--font-size-12);
  margin-top: 25px;
  padding: 20px 30px 30px;
`;

export const Heading = styled.h2`
  border-bottom: 1px solid var(--color-grey-9);
  font-family: 'Uni Sans', sans-serif;
  font-size: var(--font-size-22);
  font-weight: var(--font-weight-semibold);
  margin-bottom: 15px;
  padding: 10px 0;

  @media screen and (max-width: 899px) {
    text-align: center;
  }
`;

export const inputStyles = css`
  border: 1px solid var(--color-grey-5);
  border-bottom-left-radius: 5px;
  border-top-left-radius: 5px;
  padding: 10px;
  width: 70%;
`;

export const Button = styled.button`
  background: linear-gradient(
    to bottom,
    var(--color-sky-blue-3),
    var(--color-sky-blue-1)
  );
  border-bottom-right-radius: 5px;
  border-top-right-radius: 5px;
  box-shadow: var(--box-shadow-md);
  color: white;
  cursor: pointer;
  font-weight: var(--font-weight-bold);
  padding: 11px 15px;
  text-transform: uppercase;
  width: 30%;

  &:hover {
    box-shadow: inset 0 0 0 100px rgb(0 0 0 / 10%), var(--box-shadow-hover);
  }
`;

export const CheckboxContainer = styled.div`
  align-items: center;
  display: flex;
  font-size: var(--font-size-13);
  margin: 20px 0;
`;

export const Checkbox = styled.input`
  flex-shrink: 0;
  height: 20px;
  margin-right: 10px;
  width: 20px;
`;

export const Link = styled.a`
  color: var(--color-sky-blue-2);
  font-weight: var(--font-weight-bold);
  margin-left: 5px;
  text-decoration: none;

  &:hover {
    text-decoration: underline;
  }
`;
