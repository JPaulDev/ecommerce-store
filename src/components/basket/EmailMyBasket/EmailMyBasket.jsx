import Link from 'next/link';
import { useId, useState, memo } from 'react';
import * as Styled from './styles';

function EmailMyBasket() {
  const [email, setEmail] = useState('');
  const [isSubscribed, setIsSubscribed] = useState(false);
  const id = useId();

  const handleChangeEmail = (e) => setEmail(e.target.value);
  const handleChangeSubscribed = (e) => setIsSubscribed(e.target.checked);

  return (
    <Styled.Container>
      <Styled.Heading>Email my basket</Styled.Heading>
      <form action="https://pc-connect.co.uk/subscribe" method="post">
        <label htmlFor={`${id}-email`} hidden>
          Email
        </label>
        <Styled.Input
          id={`${id}-email`}
          placeholder="Email address"
          name="email"
          autoComplete="email"
          type="email"
          value={email}
          required
          onChange={handleChangeEmail}
        />
        <Styled.Button type="submit">Send</Styled.Button>
        <Styled.CheckboxContainer>
          <Styled.Checkbox
            id={`${id}-checkbox`}
            type="checkbox"
            name="subscribe"
            checked={isSubscribed}
            onChange={handleChangeSubscribed}
          />
          <label htmlFor={`${id}-checkbox`}>
            Also send me the latest offers and tech info, you can unsubscribe at
            any time.
          </label>
        </Styled.CheckboxContainer>
        Read our
        <Link href="/" passHref>
          <Styled.Link>privacy policy</Styled.Link>
        </Link>
      </form>
    </Styled.Container>
  );
}

export default memo(EmailMyBasket);
