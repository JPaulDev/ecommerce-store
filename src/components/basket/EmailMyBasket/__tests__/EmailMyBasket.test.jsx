import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import EmailMyBasket from '../index';

function setup(component) {
  return {
    user: userEvent.setup(),
    ...render(component),
  };
}

describe('EmailMyBasket component', () => {
  it('renders a widget that allows users to enter their email address', () => {
    const tree = renderer.create(<EmailMyBasket />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('EmailMyBasket component text input', () => {
  it('correctly records user input', async () => {
    const { user } = setup(<EmailMyBasket />);
    const input = screen.getByRole('textbox');

    expect(input).toHaveValue('');

    await user.type(input, 'test@test.com');

    expect(input).toHaveValue('test@test.com');
  });
});

describe('EmailMyBasket checkbox', () => {
  it('should be checked', async () => {
    const { user } = setup(<EmailMyBasket />);
    const checkbox = screen.getByRole('checkbox', {
      name: /also send me the latest offers/i,
    });

    expect(checkbox).not.toBeChecked();

    await user.click(checkbox);

    expect(checkbox).toBeChecked();
  });
});
