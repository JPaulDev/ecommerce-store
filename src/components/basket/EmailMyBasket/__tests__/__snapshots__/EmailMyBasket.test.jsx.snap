// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailMyBasket component renders a widget that allows users to enter their email address 1`] = `
.c0 {
  background-color: var(--color-grey-6);
  border: 1px solid var(--color-grey-9);
  border-radius: 10px;
  font-size: 0.75rem;
  margin-top: 25px;
  padding: 20px 30px 30px;
}

.c1 {
  border-bottom: 1px solid var(--color-grey-10);
  font-family: 'Uni Sans',sans-serif;
  font-size: 1.4rem;
  font-weight: 600;
  margin-bottom: 15px;
  padding: 11px 0;
}

.c2 {
  border: 1px solid #767676;
  border-bottom-left-radius: 5px;
  border-top-left-radius: 5px;
  padding: 10px;
  width: 70%;
}

.c3 {
  background: linear-gradient(to top,#1463a2,#3a7fbf);
  border-bottom-right-radius: 5px;
  border-top-right-radius: 5px;
  box-shadow: var(--shadow-button);
  color: white;
  cursor: pointer;
  font-weight: 700;
  padding: 11px 15px;
  text-transform: uppercase;
  width: 30%;
}

.c3:hover {
  background: linear-gradient(to top,#105891,#1463a2);
  box-shadow: var(--shadow-button-hover);
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 0.82rem;
  margin: 20px 0;
}

.c5 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  height: 20px;
  margin-right: 10px;
  width: 20px;
}

.c6 {
  color: var(--color-blue-1);
  font-weight: 700;
  margin-left: 5px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c6:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

@media screen and (max-width:899px) {
  .c1 {
    text-align: center;
  }
}

<div
  className="c0"
>
  <h2
    className="c1"
  >
    Email my basket
  </h2>
  <form
    action="https://pc-connect.co.uk/subscribe"
    method="post"
  >
    <label
      hidden={true}
      htmlFor=":r0:-email"
    >
      Email
    </label>
    <input
      autoComplete="email"
      className="c2"
      id=":r0:-email"
      name="email"
      onChange={[Function]}
      placeholder="Email address"
      required={true}
      type="email"
      value=""
    />
    <button
      className="c3"
      type="submit"
    >
      Send
    </button>
    <div
      className="c4"
    >
      <input
        checked={false}
        className="c5"
        id=":r0:-checkbox"
        name="subscribe"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        htmlFor=":r0:-checkbox"
      >
        Also send me the latest offers and tech info, you can unsubscribe at any time.
      </label>
    </div>
    Read our
    <a
      className="c6"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      privacy policy
    </a>
  </form>
</div>
`;
